### PR TITLE
My sinatra-twitter-oauth admited Sinatra app under sub-path

### PR DESCRIPTION
**Pull request by tsunekawa**
Hi, baroquebobcat.

I forked your repogitory and added some options. If a developer sets option "prefix", my sinatra-twitter-oauth allows a Sinatra app under sub-path to authorize with twitter.

If you like this, please pull my repository.

### CODE REVIEW

**Comment by baroquebobcat** [comment]
Thanks for the pull request. I've got some comments.

File: lib/sinatra-twitter-oauth/helpers.rb [hunk 1]
```diff
@@ -11,19 +17,19 @@ module Sinatra::TwitterOAuth
     # Redirects to login unless there is an authenticated user
     def login_required
       setup_client
-      
-      @user = ::TwitterOAuth::User.new(@client, session[:user]) if session[:user]
+      session_h = Hash[session.to_hash.map{|key,value| [key.to_sym, value]}]
```
Comment (baroquebobcat) [inline lib/sinatra-twitter-oauth/helpers.rb:20]: does `session[:user]` not work for you? it should I think. `Rack::Session::Abstract:SessionHash#[]` `to_s`s its [key arg](https://github.com/rack/rack/blob/master/lib/rack/session/abstract/id.rb#L58-61) before evaluating it, so I think this probably isn't necessary.

File: lib/sinatra-twitter-oauth.rb [hunk 2]
```diff
@@ -21,19 +22,20 @@ module Sinatra
...
+      :login_template => {:text=>'<a href="/connect">Login using Twitter</a>'},
+      :prefix => ""
     }
   
     def self.registered app  # :nodoc:
     
       app.helpers Helpers
-      app.enable :sessions
+      app.use Rack::Session::Pool, :key => 'rack.session'
       app.set :twitter_oauth_config, DEFAULT_CONFIG
```
Comment (baroquebobcat) [inline lib/sinatra-twitter-oauth.rb:33]: I'm not sure if I want to replace Cookie based sessions with pool based ones for all things, but I could see making it more flexible about session middleware.

Now, you'd need to do:

``` ruby
 register Sinatra::TwitterOAuth
disable :sessions
use Rack::Session::Pool, :key => 'rack.session'
```

which isn't very pretty.

Can you think of a better way?

We could just require `'rack.session` to be set somehow.

**Comment by baroquebobcat** [comment]
I like using settings instead of options.

File: lib/sinatra-twitter-oauth/helpers.rb (hunk 1 above)
Comment (tsunekawa) [inline lib/sinatra-twitter-oauth/helpers.rb:20]: `session[:user]` did not work in my environment ( Rack 1.3.5 on Ruby1.8.7). In `login_requied`, `session[:user]` return `nil` but `p session` return `{:user=>{ .. }, ... }`. I think this probably is bugs of `Rack::Session::Abstract::SessionHash`.

File: lib/sinatra-twitter-oauth.rb (hunk 2 above)
Comment (tsunekawa) [inline lib/sinatra-twitter-oauth.rb:33]: the reason for using `Rack::Session::Pool` is that Rack raise `Warning! Rack::Session::Cookie data size exceeds 4K. Content dropped.` when using `app.enable :sessions`. I think a update status in `session[:user]` is too big data as session data. If cutting size of session data could is possible, I want to use `app.enable :sessions`.

**Pull request closed without merging**